Automattic/jetpack
# External Media: Short-circuit requests earlier in the stack

#### Changes proposed in this Pull Request:

* Switches HTTP mock responses to running on `rest_pre_dispatch` to avoid live requests.
* Updates methods to switch from arrays to objects in those responses. 

#### Jetpack product discussion


p9dueE-3LL-p2


#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






Should be covered by automated tests. No changes in production code.

On dotcom:
* Apply D70265-code to your sandbox.
* Apply D70527-code to your sandbox.
* Run `cd bin/tests/jetpack-unit && phpunit --filter WP_Test_WPCOM_REST_API_V2_Endpoint_External_Media`
